packages/teams-js/src/public/video.ts

```diff
@@ -1,6 +1,14 @@
 import { sendMessageToParent } from '../internal/communication';
 import { registerHandler } from '../internal/handlers';
 import { ensureInitialized } from '../internal/internalAPIs';
+import { processMediaStream } from '../internal/mediaStreamUtils';
+import {
+  AllowSharedBufferSource,
```

Comment:
I pushed a commit to remove all the properties/functions from the interface,  with that change, the consumer needs to cast `video.VideoFrame` to `VideoFrame` in the `mediaStreamCallback` to access to the properties/functions of the `VideoFrame` interface. This won't take any cost because what the callback gets is in fact an instance of the `VideoFrame` interface, the SDK just pass the object it gets from the host directly to the callback. 

Speaking of how to handle the breaking change to the `VideoFrame` interface, firstly, I don't think it is likely to happen, it is an open spec and already implemented by many products, a breaking change would cost the world a lot. Secondly, we are running on webview2, when a breaking change does happen and webview2 decides to follow the change, there is still a chance to convert the change in the `getTextureStream` API in webview2. But again, breaking change to a open spec is unlikely to happen in my opinion.

Reply:
Thanks, that makes sense. I mostly just wanted to make sure you were aware of how hard it would be to make any meaningful change to this quickly if there ever *is* a breaking change. I agree with you that it's unlikely.